compiler/rustc_trait_selection/src/traits/wf.rs

```diff
@@ -339,17 +331,17 @@ impl<'a, 'tcx> WfPredicates<'a, 'tcx> {
                 })
                 .filter(|(_, arg)| !arg.has_escaping_bound_vars())
                 .map(|(i, arg)| {
-                    let mut new_cause = cause.clone();
+                    let mut cause = traits::ObligationCause::misc(self.span, self.body_id);
```

Comment:
Out of curiosity, could we keep the original `clause` as part of a chain? I wonder if this would improve the output.